### ISSUE TITLE
MedianCutQuantization exported colors seem off compared to UniformQuantization

### ISSUE THREAD

**Issue by Jantho1990** [jegor377/godot-gdgifexporter#15]
I did a GIF export following the sample code in the README, based on a series of images looking like the below:
![000](https://user-images.githubusercontent.com/20214977/125463929-458af46a-a491-4cc1-909c-27f4b42f8c28.png)

The export, using MedianCutQuantization, looks like this:
![test2](https://user-images.githubusercontent.com/20214977/125464159-2887a304-f78b-4aad-81b9-ffe7427cb431.gif)

I then tried using UniformQuantization, and it looks like this:
![test3](https://user-images.githubusercontent.com/20214977/125464264-fd33fcb2-a1df-4765-8fdb-21b1793ee34c.gif)

I'm not an expert on compression, but the first result seems noticeably worse than the second. Is that expected behavior for those two quantization implementations?

**Comment by jegor377** [comment]
Sorry to respond so late. My implementation of median cut is not always producing 256 colors out of the pallete. This is because sometimes a bucket is empty and when it is empty, then it can't be divided into two new buckets. I tried to divide the non-empty buckets but this resulted in palette that does not have 256 colors but for example 325. This is why I just made the algorithm to just divide the palette 8 times and it mostly just works. Because of this the algorithm may not generate enough colors for some limited palettes. Uniform in the other hand always produces 256 colors and for simple palletes it's mostly correct.
